src/main/java/core/basesyntax/service/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service;
+
+import core.basesyntax.Colors;
+import core.basesyntax.Figure;
+import core.basesyntax.Figures;
+import core.basesyntax.entity.Circle;
+import core.basesyntax.entity.IsoscelesTrapezoid;
+import core.basesyntax.entity.Rectangle;
+import core.basesyntax.entity.RightTriangle;
+import core.basesyntax.entity.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_VALUE = 10;
+    private static final int MAX_VALUE = 100;
+    private Random random = new Random();
+
+    public Figure getRandom() {
+        int randomFigureNumber = random.nextInt(Figures.values().length);
+
+        switch (Figures.values()[randomFigureNumber]) {
+            case SQUARE:
+                return new Square(getDefaultValue());
```

Comment:
Use ColorSupplier in this class and pass a random color to the constructor. Also, you ignored my comment. Put creation of random shapes into private methods. For example like this: 
public Figure getRandomSquare() {
        Square square = new Square(getDefaultValue());
        square.setColor(colorSupplier.getRandomColor());
        return square;
    }